docs/MigrationGuide.md

```diff
@@ -0,0 +1,118 @@
+- No need for SLF4J dependency and the usage of the native Android logging system.
+- Respecting the Android lifecycle to stop polling and sending metrics in the background.
+- The new SDK respects the minimum Android API level 21, but we recommend API level 23.
+- Monitoring network connectivity to avoid unnecessary polling (requires API level 23 or above).
+
+## Overview
+
+The new SDK introduces several changes and improvements, including API modifications and a new initialization process.
+
+## Step-by-Step Migration
+
+### 1. Update Gradle Dependency
+
+First, update the dependency in your `build.gradle` file:
+
+```gradle
+dependencies {
+    // Remove the old SDK
+    implementation 'io.getunleash:unleash-android-proxy-sdk:0.5.0'
+
+    // Add the new SDK
+    implementation 'io.getunleash:unleash-android:$version'
+}
+```
+
+### 2. Update the initialization code
+We won't cover all the details here as most of the configuration can be set using the builders fluent methods. However, the main difference is that the new SDK requires an `Application` context to be passed to the `Unleash` constructor. This is necessary to monitor the network connectivity and respect the Android lifecycle (usually it can be injected if using hilt with `@ApplicationContext context`).
+ 
+#### Unleash context initialization
+The main differences are:
+1. The application name is no longer configurable through the context as it is unique across the application. The `appName` should be set using the `UnleashConfig` builder.
+2. The instance ID is no longer configurable. The SDK will generate a unique instance ID for each instance.
+3. Update the import statements to use the new SDK classes.
+
+```kotlin
+val unleashContext = UnleashContext.newBuilder()
+    // .appName("unleash-android") // remove this line
+    // .instanceId("main-activity-unleash-demo-${Random.nextLong()}") // remove this line
+    .userId("unleash_demo_user")
+    .sessionId(Random.nextLong().toString())
+    .build()
+```
+
+#### Unleash configuration
+The main differences are:
+1. Metrics are enabled by default.
+2. App name is now a mandatory parameter to the builder.
+3. Instance id is no longer configurable.
+4. The polling mode is now a polling strategy with a fluent API.
+5. The metrics interval is now part of the metrics strategy with a fluent API.
+
+**Old SDK**
+```kotlin
+UnleashConfig.newBuilder()
+    .appName("unleash-android")
+    .instanceId("unleash-android-${Random.nextLong()}")
+    .enableMetrics()
+    .clientKey("xyz")
+    .proxyUrl("https://eu.app.unleash-hosted.com/demo/api/frontend")
+    .pollingMode(
+        PollingModes.autoPoll(
+            autoPollIntervalSeconds = 15
+        ) {
+
+        }
+    )
+    .metricsInterval(5000)
+    .build()
+```
+
+**New SDK**
+```kotlin
+UnleashConfig.newBuilder("unleash-android")
+    .clientKey("xyz")
+    .proxyUrl("https://eu.app.unleash-hosted.com/demo/api/frontend")
```

Comment:
do we still use `proxyUrl`? What if you use edge? Or Unleash directly (as in this example)? Is it too late to rename that property?

Reply:
I think for consistency with other sdks proxyUrl is the right thing here. Maybe in the future we could replace this, but I'm not sure if edge is best, I think I rather use frontend or frontendApiUrl... but I want to keep that out of this PR otherwise we keep having SDKs that drift apart.

Reply:
I wasn't sure about this. I suspect that 5 levels of headers is quite a lot for such short guide. Maybe I should organize this differently but the bold text seems to be fine IMO

Reply:
Yes , sir 🫡